### ISSUE TITLE
Collecting requests from multiple GET requests from various coroutines

### ISSUE THREAD

**Issue by Aditya23456** [keplerproject/copas#65]
Hi,

I am just trying to make multiple GET requests to some urls and collect responses to parse the results. The example shown in High level requests matches what I need but it prints the result. Is there any way to collect the results of all the coroutines into a table. I tried doing this-

all_data={}
for _, host in ipairs(list) do 
results=copas.addthread(handler, host) 
table.insert(all_data,results)
end
copas.loop()
 
But all_data ends up being a collection of threadlists. I am not sure how to get the results from this thread. I see in general coroutines, result is retrieved by coroutine.resume but I don't see any copas.resume() method? 

Also, for my purpose, i don't have any server implementation and just require an asynchronous client requesting different URLs. I feel for my application, I can just implement a simple coroutine example given in LUA coroutine tutorial instead of copas library. I am just curious if using copas for my application would add enough robustness/error handling so that I can overlook the overhead added by copas library or is copas library primarily used for server handling applications. It would be great if overhead can be estimated.

Thanks,
Aditya

**Comment by Tieske** [comment]
@Aditya23456 `addthread` will return the newly created coroutine. You should store the result at the end of your handler function.

```lua
local all_data = {}  -- create an upvalue

local handler = function(...)
  -- do the usual stuff here

  -- when done, and the result is complete, store it
  table.insert(all_data, my_new_result)
end

for _, host in ipairs(list) do
  results=copas.addthread(handler, host)
  -- by now, we created the handlers, but they aren't running yet, let alone have a result
end
copas.loop()  -- will exit when all handlers have finished

-- since all handlers have completed now, the results should be in the list
```

**Comment by Aditya23456** [comment]
Hii.. 

Thank you very much for your reply. I have another issue. I am using a lua lanes multithreaded library to launch 3 threads. One of the thread uses copas to make async calls to urls. At my handler function, I parse and update an SQLLite database with the result of URL response. I don't know why but my other threads can access SQLite database without any issue but the DB transactions done in the handler gets stuck with "LuaSQL: database is locked" error. I made sure my database connection and closing are done after the asynchttp statement so that entire transaction is done by current handler and then control is yielded to the other async request handler. Am I still doing something wrong? I am planning to send all my results by storing in upvalue to the main thread and then do the SQLite transaction for each response(I am not sure if it works but this increases time complexity). Do you have any idea how I can get through this ?

Thank you once again,
Regards,
Aditya

**Comment by Tieske** [comment]
I think this question should better be posted on the Lua mailing list, preferably with some code that shows the issue you're having.

see https://www.lua.org/lua-l.html

**Comment by Aditya23456** [comment]
Hi.. Thanks for replying. I have one small issue.. If one of my url request fails to respond it takes 60 seconds to timeout. So I set it to 2 seconds in this way at the beginning of my code but still it doesn't get overriden and waits for 60 seconds-

copashttp = require("copas.http")
copashttp.TIMEOUT = 2
asynchttp=copashttp.request

and then used asynchttp object for making multiple url requests.
I tried to follow copas.http sourcecode and feel .TIMEOUT is not implemented for the client scenario. Is it so or I have a wrong implementation? Is there any way I can timeout by 2 seconds?
